### PR TITLE
Changed language of the PR template to enable auto-linking to issues.

### PR DESCRIPTION
**Pull request by JoaoRodrigues**
As briefly discussed in another PR (#3208), using the 'Resolves ...' syntax at the end of the PR (closing keywords) tells github to automatically close that issue when the PR is merged.

- [x] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [x] I have read the ``CONTRIBUTING.rst`` file, have run ``flake8`` locally, and
understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [x] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

### CODE REVIEW

**Comment by peterjc** [comment]
Any additional voices in favour or against?

**Comment by chris-rands** [comment]
Does `Resolves issue #...` definitely trigger the keyword rather than `Resolves #...`? Also is there a preference for `Resolves` over the `Closes` keyword? [docs link](https://docs.github.com/en/enterprise/2.16/user/github/managing-your-work-on-github/closing-issues-using-keywords)

Many PRs only address a small part of an Issue, so closing isn't alway appropriate- I wonder if Issues may be closed accidentally

**Comment by peterjc** [comment]
During the recent triage during the BCC2020 CoFest and since, we found a lot of forgotten issues where a pull request was applied, but the matching issue left open (I didn't count, easily a dozen issues - maybe even half of those I closed recently). That clearly has been a problem. If we change the template and swing to the other extreme, we will sometimes have issues being closed prematurely by only a partial fix, but I would hope the issue reporter would notice and reopen?

Or, we might try something like: ``This addresses / closes #... (delete as appropriate).`` but that is perhaps overcomplicated?

**Comment by chris-rands** [comment]
If you think it's better to accidentally close some issues than accidentally leave some open, then go for it

In terms of the semantics, I think `Closes` is more explicit than `Resolves`, but it's subjective I guess

**Comment by JoaoRodrigues** [comment]
For everyone's info, [here](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) is the link to the closing keywords documentation. The options are: 
```
    close
    closes
    closed
    fix
    fixes
    fixed
    resolve
    resolves
    resolved
```

I personally prefer the word 'resolves' because it sounds like you solved the issue, and not just closed the issue. You can close an issue without solving it. But it's semantics at this point so I'm fine with whatever everyone else prefers.

As for partial solutions to issues, I think it's not that much to ask the person that merges the commit onto the master branch to double check if that is the case and just edit the PR message accordingly (e.g. to remove the keyword).

**Comment by peterjc** [comment]
I have no strong preference on the verb. Fixes is perhaps simpler than resolves - so I'd probably go for fixes or closes.

In the case of partial fixes, manually reopening the issue is also easy enough for the person doing the merge.

**Comment by JoaoRodrigues** [comment]
Closes sounds good then, since the idea is to keep it for PRs that do close issues.

**Comment by JoaoRodrigues** [comment]
Changed the wording and added a small comment to let users know where to write about the PR.

**Comment by JoaoRodrigues** [comment]
TODO: The PR text also needs to be updated to reflect the `pre-commit` change instead of mentioning flake8.

**Comment by JoaoRodrigues** [comment]
@chris-rands do you find this current version better?

**Comment by chris-rands** [comment]
Yes, thanks, as I said before, I assume `Closes issue #...` syntax triggers the closure as well as the examples in the docs with `Closes #...`?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,15 +1,18 @@
...
+- [ ] I have read the ``CONTRIBUTING.rst`` file, have run ``pre-commit`` locally,
+and understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
 tests and style checks pass with these changes.
 
 - [ ] I have added my name to the alphabetical contributors listings in the files
 ``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
 already, or do not wish to be listed. (*This acknowledgement is optional.*)
+
+<!--- Briefly describe the changes included in this pull request below
+ !--- starting with the words 'Closes issue #... --->
```
Comment (peterjc) [inline .github/PULL_REQUEST_TEMPLATE.md:16]: How about:

``starting with 'Closes issue #...' if appropriate``

Comment (JoaoRodrigues) [reply .github/PULL_REQUEST_TEMPLATE.md:16]: Sounds good!

Comment (JoaoRodrigues) [reply .github/PULL_REQUEST_TEMPLATE.md:16]: Got it!

**Comment by JoaoRodrigues** [comment]
@chris-rands Good catch! The documentation says "Keywords followed by issue number".

**Comment by peterjc** [comment]
The comment needs to change to match; other than that good to go.

**Comment by JoaoRodrigues** [comment]
There we go!

**Comment by JoaoRodrigues** [comment]
Done! Thank you both!

**Pull request merged**